test/integration/camera_sensor.cc

```diff
@@ -2040,4 +2040,198 @@ TEST_F(CameraSensor, SetCompositorNames)
+    camSensorBarrelLegacy->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount3, img3,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c4 =
+    camSensorBarrelNew->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount4, img4,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+
+  // Get some images
+  while (
+    imageCount < 10 || imageCount2 < 10 ||
+    imageCount3 < 10 || imageCount4 < 10)
+  {
+    common::Time::MSleep(10);
+  }
+
+  unsigned int colorSum1 = 0;
+  unsigned int colorSum2 = 0;
+  unsigned int colorSum3 = 0;
+  unsigned int colorSum4 = 0;
+  const unsigned int middleRow = height / 2;
+  for (unsigned int x = 0; x < width*3; x+=3)
+  {
+    unsigned int r1 = img[(middleRow*width*3) + x];
+    unsigned int g1 = img[(middleRow*width*3) + x + 1];
+    unsigned int b1 = img[(middleRow*width*3) + x + 2];
+    colorSum1 += r1 + g1 + b1;
+    unsigned int r2 = img2[(middleRow*width*3) + x];
+    unsigned int g2 = img2[(middleRow*width*3) + x + 1];
+    unsigned int b2 = img2[(middleRow*width*3) + x + 2];
+    colorSum2 += r2 + g2 + b2;
+    unsigned int r3 = img3[(middleRow*width*3) + x];
+    unsigned int g3 = img3[(middleRow*width*3) + x + 1];
+    unsigned int b3 = img3[(middleRow*width*3) + x + 2];
+    colorSum3 += r3 + g3 + b3;
+    unsigned int r4 = img4[(middleRow*width*3) + x];
+    unsigned int g4 = img4[(middleRow*width*3) + x + 1];
+    unsigned int b4 = img4[(middleRow*width*3) + x + 2];
+    colorSum4 += r4 + g4 + b4;
+  }
+
+  // Check that the legacy mode distorts the pincushion images less
+  EXPECT_GT(colorSum1, colorSum2+800);
+  // Check that there is a good difference in the barrel images
+  // this difference is largely caused by the edges of the new
+  // distortion model which appear gray when they have no value
+  // and gray (192) has a much lower value than white (765)
+  EXPECT_GT(colorSum3, colorSum4+20000);
+
+  // Check that the corners contain rendered pixels.
+  // Specifically, the corners in each image should be white and
+  // have a value of 765=255*3 since the background of the environment
+  // is white
+  unsigned int cornerColorSumImg1 = img[0] + img[1] + img[2];
+  unsigned int cornerColorSumImg2 = img2[0] + img2[1] + img2[2];
+  unsigned int cornerColorSumImg3 = img3[0] + img3[1] + img3[2];
+  EXPECT_EQ(cornerColorSumImg1, 765u);
+  EXPECT_EQ(cornerColorSumImg2, 765u);
+  EXPECT_EQ(cornerColorSumImg3, 765u);
+  // Check that this image is not cropped and that the corner pixel
+  // has the gray value assigned to unrendered pixels
+  unsigned int cornerColorSumImg4 = img4[0] + img4[1] + img4[2];
+  EXPECT_EQ(cornerColorSumImg4, 192u);
+
+  auto getFirstColIdxOfMineCart = [middleRow, width](const unsigned char* img)
```

Comment:
I just noticed that clang warns that you don't need to capture `middlerow` and `width`

* https://build.osrfoundation.org/job/gazebo-ci-pr_any-homebrew-amd64/2443/clang/folder.-566429772/fileName.-806121611/

I'm guessing it is because they are `const` integers?